src/main/kotlin/creator/custom/types/IntegerCreatorProperty.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Minecraft Development for IntelliJ
+ *
+ * https://mcdev.io/
+ *
+ * Copyright (C) 2024 minecraft-dev
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, version 3.0 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.demonwav.mcdev.creator.custom.types
+
+import com.demonwav.mcdev.creator.custom.PropertyDerivation
+import com.demonwav.mcdev.creator.custom.TemplatePropertyDescriptor
+import com.demonwav.mcdev.creator.custom.model.HasMinecraftVersion
+import com.demonwav.mcdev.platform.sponge.util.SpongeVersions
+import com.demonwav.mcdev.util.MinecraftVersions
+import com.demonwav.mcdev.util.SemanticVersion
+import com.intellij.ide.util.projectWizard.WizardContext
+import com.intellij.openapi.observable.properties.PropertyGraph
+import com.intellij.ui.dsl.builder.COLUMNS_LARGE
+import com.intellij.ui.dsl.builder.Panel
+import com.intellij.ui.dsl.builder.bindIntText
+import com.intellij.ui.dsl.builder.columns
+
+class IntegerCreatorProperty(
+    graph: PropertyGraph,
+    descriptor: TemplatePropertyDescriptor,
+    properties: Map<String, CreatorProperty<*>>
+) : SimpleCreatorProperty<Int>(graph, descriptor, properties) {
+
+    override fun createDefaultValue(raw: Any?): Int = raw as? Int ?: 0
+
+    override fun serialize(value: Int): String = value.toString()
+
+    override fun deserialize(string: String): Int = string.toIntOrNull() ?: 0
```

Comment:
Maybe these `deserialize` functions could return a nullable value, so that if the jsons are invalid a warning could be logged? (and then `createDefaultValue` could be used)

Reply:
This is one of the places I couldn't really figure out properly as it was written very early.
Your solution sounds good

Reply:
So I think I'll keep it like that because `createDefaultValue` can take anything, usually it's a String, but it can be an Int, Boolean or even a Map if the property supports it.

The serialization methods are meant only for remembering values and have to be from/to Strings.

Deserialize can always fallback to the default value, I leave it up to the implementation to decide.